### PR TITLE
Draft: Add Japanese translation

### LINKED ISSUE #59: various languages
I would like to add a language
Languages ​​that need to be added
Japanese

Also, if I can add it myself, please let me know how to do it.

It would be nice to have a function that allows you to edit the displayed characters yourself

### PR DESCRIPTION
**Pull request by hilot06**
This request is to add an Japanese translation.
The translation file is in [src/app/i18n/locale/ja-jp.ts](https://github.com/hilot06/nxapi/blob/i18n/src/app/i18n/locale/ja-jp.ts).

I couldn't translate all because I don't know meaning of the specific words (ex. `NSA` or/and `Coral` ).
If you let me know about meaning, I'll fix it. (... and may mark ready for review)

-----

Thank you for your time to look this PR!
hilot06

P.S. Ah, I made an typo in a commit name...

- Closes #59.

### CODE REVIEW

**Comment by hilot06** [comment]
To Japanese Translators,

タイプミスをしていたり、そもそも間違っていたりなどなど色々不備がある可能性があります。
拙作の[日本語ファイル](https://github.com/hilot06/nxapi/blob/i18n/src/app/i18n/locale/ja-jp.ts)と、元々こちらのプロジェクトに入っている[英語ファイル](https://github.com/samuelthomas2774/nxapi/blob/i18n/src/app/i18n/locale/en-gb.ts)を比較頂けると幸いです。
修正点があれば恐れ入りますがCommit頂けると嬉しいです。このPull Requestへのコメントのみでも泣いて喜びます…！
なにとぞよろしくお願いいたします。

**Review by samuelthomas2774 — CHANGES_REQUESTED** [review]
Thank you for adding this! I've added comments for some keys that need checking; [I've also added some extra keys for things I missed since this was opened](https://github.com/samuelthomas2774/nxapi/compare/b9b9f6a8d74d1ba24ef8b84fc502d04985e64ed7..f8ab3721714849a77b9feb056aa291bbacd91019#diff-6439e40dd9e95cc1adf4effe25ea70a75c5a52dc0e04d831b9416f1a50da8d54).

> I couldn't translate all because I don't know meaning of the specific words (ex. NSA or/and Coral ).

NSA (Network Service Account), Coral, Moon, etc. are internal terms used by Nintendo, these don't need to be translated.

File: src/app/i18n/index.ts [hunk 1]
```diff
@@ -27,6 +27,13 @@ export const languages = {
             ['sarayalth', 'https://github.com/sarayalth'],
         ],
     },
+    'ja-JP': {
+        name: 'Japanese',
```
Comment (samuelthomas2774) [inline src/app/i18n/index.ts:31]: The name here is shown in the about this app window using `app:translation_credits` (e.g. using the non-localised name in #66:), so this should also be localised.

Screenshot 2023-06-02 at 23 24 48

(Thanks @Sarayalth for pointing this out.)

**Review by samuelthomas2774 — CHANGES_REQUESTED** [review]

File: src/app/i18n/locale/ja-jp.ts [hunk 2]
```diff
@@ -0,0 +1,352 @@
+import { CREDITS_NOTICE, LICENCE_NOTICE } from '../../../common/constants.js';
+
+export const app = {
+    default_title: 'Nintendo Switch Online',
+
+    licence: LICENCE_NOTICE,
+    credits: CREDITS_NOTICE,
+    translation_credits: 'Japanese translation by hilot06.',
```
[the remark this answers is not in the document]
Comment (hilot06) [reply src/app/i18n/locale/ja-jp.ts:8]: OK. I'll change it right away.

File: src/app/i18n/locale/ja-jp.ts [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+        share: '共有する',
+    },
+};
+
+export const preferences_window = {
+    title: '設定',
+
+    startup: {
+        heading: 'スタートアップ',
+        login: 'Windowsにログインした時に開く',
```
[the remark this answers is not in the document]
Comment (hilot06) [reply src/app/i18n/locale/ja-jp.ts:216]: Ah, OK. So I'll change it to 'OS'.

File: src/app/i18n/locale/ja-jp.ts [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+
+        game_play_time_h: 'プレイ時間: $t(friend.hours, {"count": {{hours}}})',
+        game_play_time_hm: 'プレイ時間: $t(friend.hours, {"count": {{hours}}}), $t(friend.minutes, {"count": {{minutes}}})',
+        game_play_time_m: 'プレイ時間: $t(friend.minutes, {"count": {{minutes}}})',
+        hours_one: '{{count}}時間',
+        hours_other: '{{count}}時間',
+        minutes_one: '{{count}}分',
+        minutes_other: '{{count}}分',
+
+        presence_inactive: 'オンライン (ゲーム未起動)',
```
Comment (hilot06) [inline src/app/i18n/locale/ja-jp.ts:81]: I translated "Offline (concole online)" to "Online (no game has launched)".

File: src/app/i18n/locale/ja-jp.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (hilot06) [reply src/app/i18n/locale/ja-jp.ts:216]: Ah OK. But I think users will confuse because they'll think about NSA.
So I'll use 'OS' instead.

**Comment by hilot06** [comment]
Thanks a lot to Samuel and Sarah!
I've changed some bad translations and added `na_auth`. Please checked it out.
Also I've checked all translations again, and fix some translations.

**Comment by hilot06** [comment]
i knew i can request a review 20 minutes ago...
it's difficult to operate github...

**Comment by nennneko5787** [comment]
I have revised the text to fit this document, what do you think?
https://www.jtf.jp/pdf/jtf_style_guide.pdf

https://github.com/nennneko5787/nxapi/blob/i18n/src/app/i18n/locale/ja-jp.ts

**Pull request closed without merging**